trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/remote/RemoteS3Config.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.remote;
+
+import io.airlift.configuration.Config;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.Optional;
+
+public class RemoteS3Config
+{
+    private boolean https = true;
+    private String domain = "amazonaws.com";
+    private Optional<Integer> port = Optional.empty();
+
+    @Config("remoteS3.https")
+    public RemoteS3Config setHttps(Boolean https)
+    {
+        this.https = https;
+        return this;
+    }
+
+    @Config("remoteS3.domain")
+    public RemoteS3Config setDomain(String s3Domain)
+    {
+        this.domain = s3Domain;
+        return this;
+    }
+
+    @Config("remoteS3.port")
+    public RemoteS3Config setPort(Integer port)
```

Comment:
Use lowercase, native `int`.

Reply:
Port is used as an optional attribute

Reply:
`setPort()` would only be called if someone wants to set it. It's `empty()` by default.

Reply:
This is used for testing default configuration - https://github.com/trinodb/aws-proxy/pull/116/files#diff-f7bfe83b2cf635f8e16cad59f8d150997c4b42213844109499532ddc54fad3d4R31

Reply:
I wondered the same thing when I saw this part @Randgalt , but as @pranavr12 said it seems like this is what other plugins do on the Trino side too. For instance, https://github.com/trinodb/trino/blob/0c672b62443643c1eacdf5153d72460736373c31/plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClientConfig.java#L350-L355

No objection either way, just curious as to what the "idiomatic Airlift" way is here

Reply:
I'm OK with either. I avoid boxed types whenever possible.